homeassistant/components/discord/notify.py

```diff
@@ -78,15 +88,28 @@ async def on_ready():
                     ) or discord_bot.get_user(channelid)
 
                     if channel is None:
-                        _LOGGER.warning("Channel not found for id: %s", channelid)
+                        _LOGGER.warning("Channel not found for ID: %s", channelid)
                         continue
                     # Must create new instances of File for each channel.
                     files = None
                     if images:
                         files = []
                         for image in images:
                             files.append(discord.File(image))
-                    await channel.send(message, files=files)
+                    if ATTR_EMBED in data:
+                        embed = discord.Embed(**embedding)
+                        if ATTR_EMBED_FIELDS in embedding:
+                            for field_num, field_name in enumerate(fields):
+                                embed.add_field(**fields[field_num])
+                        if ATTR_EMBED_FOOTER in embedding:
+                            embed.set_footer(**embedding[ATTR_EMBED_FOOTER])
+                        if ATTR_EMBED_AUTHOR in embedding:
+                            embed.set_author(**embedding[ATTR_EMBED_AUTHOR])
+                        if ATTR_EMBED_THUMBNAIL in embedding:
+                            embed.set_thumbnail(**embedding[ATTR_EMBED_THUMBNAIL])
+                        await channel.send(message, files=files, embed=embed)
```

Comment:
That message, that will be above the embedded content.
I wonder if it makes sense to use the "description" key of the embed object for that instead.

<https://leovoel.github.io/embed-visualizer/>